mainnet_contract/contracts/LiteCheckpoint.sol

```diff
@@ -0,0 +1,434 @@
+        saveEpoch(header0);
+        //for commit header0 util
+        if (headers.length > 1) {
+            bytes32 blockHash = keccak256(header0);
+            commitHeader(blockHash, sliceBytes(headers, 1));
+        }
+    }
+
+    function commitHeaderByNumber(
+        uint256 number,
+        bytes[] memory headers
+    ) external {
+        bytes32 epochHash = heightTree[uint64(number)];
+        commitHeader(epochHash, headers);
+    }
+
+    /*
+     * @description commit header
+     * 1. (Conditional) Update Committed Status for ancestor blocks
+     * @param epochHash the gap/epoch block hash that need to continue commit ，util gap/epoch block hash committed
+     * @param headers list of rlp-encoded block headers.
+     */
+    function commitHeader(bytes32 epochHash, bytes[] memory headers) public {
+        require(
+            headers.length > 0,
+            "commitHeader : Headers length must be greater than 0"
+        );
+
+        require(epochHash != 0, "Error epoch hash");
+        bytes32 parenHash = unCommittedLastHash[epochHash];
+        require(parenHash != 0, "EpochHash not found, may not have been saved");
+
+        UnCommittedHeaderInfo memory uc = getUnCommittedHeader(epochHash);
+
+        uint64 sequence = uc.sequence;
+        uint64 lastRoundNum = uc.lastRoundNum;
+        uint64 lastNum = uc.lastNum;
+
+        for (uint256 x = 0; x < headers.length; x++) {
+            HeaderReader.ValidationParams memory validationParams = HeaderReader
+                .getValidationParams(headers[x]);
+            bytes32 blockHash = keccak256(headers[x]);
+            checkSig(validationParams);
+
+            if (validationParams.parentHash != parenHash) {
+                revert("Invalid Block Num Sequence");
+            }
+
+            if (validationParams.roundNumber == lastRoundNum + 1) {
+                unchecked {
+                    sequence++;
+                }
+            } else {
+                sequence = 0;
+            }
+
+            lastNum = uint64(uint256(validationParams.number));
+            lastRoundNum = uint64(validationParams.roundNumber);
+            parenHash = blockHash;
+            if (sequence >= 3) {
+                break;
+            }
+        }
+
+        if (sequence >= 3) {
+            headerTree[epochHash] = clearLowest(headerTree[epochHash], 64);
+            headerTree[epochHash] |= block.number;
+            latestFinalizedBlock = epochHash;
+            delete unCommittedTree[epochHash];
+            delete unCommittedLastHash[epochHash];
+        } else {
+            unCommittedTree[epochHash] =
+                (uint256(sequence) << 128) |
+                (uint256(lastRoundNum) << 64) |
+                uint256(lastNum);
+            bytes32 blockHash = keccak256(headers[headers.length - 1]);
+            unCommittedLastHash[epochHash] = blockHash;
+        }
+    }
+
+    function saveEpoch(bytes memory header) private {
```

Comment:
I find that this function means save epochSwitch/gapPlusOne. Can you change name to `saveEpochOrGap`?

Reply:
https://github.com/XinFinOrg/XDC-Subnet/pulls#discussion_r1299152700 right

Reply:
> I find that this function means save epochSwitch/gapPlusOne. Can you change name to `saveEpochOrGap`?

fixed